gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java

```diff
@@ -16,134 +16,136 @@
 
+import android.content.SharedPreferences;
 import androidx.navigation.NavDirections;
-import com.google.android.gnd.model.Project;
-import com.google.android.gnd.model.TermsOfService;
-import com.google.android.gnd.model.User;
-import com.google.android.gnd.repository.ProjectRepository;
 import com.google.android.gnd.repository.TermsOfServiceRepository;
 import com.google.android.gnd.repository.UserRepository;
 import com.google.android.gnd.system.auth.FakeAuthenticationManager;
 import com.google.android.gnd.system.auth.SignInState;
 import com.google.android.gnd.system.auth.SignInState.State;
-import com.google.android.gnd.ui.common.EphemeralPopups;
 import com.google.android.gnd.ui.common.Navigator;
 import com.google.android.gnd.ui.home.HomeScreenFragmentDirections;
 import com.google.android.gnd.ui.signin.SignInFragmentDirections;
-import dagger.hilt.android.testing.BindValue;
 import dagger.hilt.android.testing.HiltAndroidTest;
-import io.reactivex.Completable;
-import io.reactivex.Flowable;
-import io.reactivex.Maybe;
-import java8.util.Optional;
+import io.reactivex.observers.TestObserver;
+import java.util.NoSuchElementException;
 import javax.inject.Inject;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.Mockito;
 import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowToast;
 
 @HiltAndroidTest
 @RunWith(RobolectricTestRunner.class)
 public class MainViewModelTest extends BaseHiltTest {
 
-  private static final TermsOfService TEST_TERMS_OF_SERVICE = FakeData.TEST_TERMS_OF_SERVICE;
-  private static final Optional<Project> TEST_ACTIVE_PROJECT = Optional.of(FakeData.TEST_PROJECT);
-  private static final User TEST_USER = FakeData.TEST_USER;
-
-  @BindValue @Mock EphemeralPopups mockPopups;
-  @BindValue @Mock Navigator mockNavigator;
-  @BindValue @Mock ProjectRepository mockProjectRepository;
-  @BindValue @Mock TermsOfServiceRepository mockTosRepository;
-  @BindValue @Mock UserRepository mockUserRepository;
-
   @Inject FakeAuthenticationManager fakeAuthenticationManager;
   @Inject MainViewModel viewModel;
+  @Inject Navigator navigator;
+  @Inject SharedPreferences sharedPreferences;
+  @Inject TermsOfServiceRepository tosRepository;
+  @Inject UserRepository userRepository;
+
+  private TestObserver<NavDirections> navDirectionsTestObserver;
 
   @Before
   public void setUp() {
     // TODO: Add a test for syncFeatures
-    when(mockProjectRepository.getActiveProject()).thenReturn(Flowable.just(TEST_ACTIVE_PROJECT));
-
     super.setUp();
+
+    // Subscribe to navigation requests
+    navDirectionsTestObserver = navigator.getNavigateRequests().test();
+  }
+
+  private void setupUserPreferences() {
+    sharedPreferences.edit().putString("foo", "bar").apply();
+  }
+
+  private void assertUserPreferencesCleared() {
+    assertThat(sharedPreferences.getAll()).isEmpty();
+  }
+
+  private void assertUserSaved() {
```

Comment:
Nit: Methods that execute the unit under test are not only asserting, they're actually tests themselves. The easiest way to make this clear would be to rename them with something like "verify*"

Reply:
Done